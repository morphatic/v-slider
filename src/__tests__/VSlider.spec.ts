import Vue from 'vue'
// Utils
import { mount, MountOptions, Wrapper } from '@vue/test-utils'
import { inspect } from 'util' // okay to delete this line if you don't use node inspect

// Mocks
const vuetifyMocks = {
  $vuetify : {
    theme: {
      currentTheme: {
        error: '#ff0000',
      },
      dark: false,
    },
    lang: {
      t: (val: string) => val,
    },
  },
}

// Component to be tested
import MorphaticVuetifyExtensions, { VSlider } from '..'

// Data necessary for tests
const foo = 'bar'
const messages = []

describe('VSlider', () => {
  describe('installer', () => {
    it('should register the your-component component', () => {
      Vue.use(MorphaticVuetifyExtensions)
      expect(Vue.options.components['your-component']).toBeTruthy()
    })
  })

  describe('component', () => {
    type Instance = InstanceType<typeof VSlider>
    let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>
  
    beforeEach(() => {
      mountFunction = (options?: MountOptions<Instance>) => {
        return mount(VSlider, {
          mocks: {
            ...vuetifyMocks,
          },
          ...options,
        })
      }
    })
  
    describe('initialization', () => {
      beforeEach(() => {
        // reset messages
        messages.length = 0
        // capture console messages to prevent them from cluttering the terminal
        const capture = m => { messages.push(m.toString()) }
        global.console.error = jest.fn(capture)
        global.console.warn = jest.fn(capture)
        global.console.log = jest.fn(capture)
      })
  
      it('should render component and match snapshot', () => {
        const wrapper = mountFunction({
          propsData: { foo },
        })
        // replace the auto-generated `id` with one that matches
        const html = wrapper.html().replace(/div id="input-\d+"/, 'div id="input-1"')
        expect(html).toMatchSnapshot()
      })
        
      it('should have a property called `foo`', () => {
        const wrapper = mountFunction({ propsData: { foo }, })
        expect(wrapper.vm.foo).toBeDefined()
        expect(wrapper.vm.foo).toBe('bar')
      })
    })
  
    describe('internal functions and events', () => {
      let wrapper
      beforeEach(() => {
        // capture console.warn
        const capture = m => { messages.push(m.toString()) }
        global.console.warn = jest.fn(capture)
        wrapper = mountFunction({
          propsData: { foo }
        })
      })
  
      it('bar() should return "baz"', () => {
        const result = wrapper.vm.bar()
        expect(result).toBe('baz')
      })  
    })
  })
})
