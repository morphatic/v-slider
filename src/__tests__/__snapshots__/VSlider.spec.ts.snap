// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSlider component initialization should render component and match snapshot 1`] = `
<div class="v-input theme--light v-text-field">
  <div class="v-input__control">
    <div class="v-input__slot">
      <div class="v-text-field__slot">
        <input id="input-1"
               type="text"
        >
      </div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div"
                               name="message-transition"
                               class="v-messages__wrapper"
        >
        </transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
